src/containers/Petitions.js

```diff
@@ -1,26 +1,76 @@
 import React from 'react';
 import Helmet from 'react-helmet';
 import { connect } from 'react-redux';
-import { fetchPetitions } from 'actions/PetitionsActions';
+import { withRouter } from 'react-router';
+import { petitionsPath } from 'helpers/petitionUrls';
+import { fetchPetitions, updateCityFilterValue } from 'actions/PetitionsActions';
 import settings from 'settings';
 import Petitions from 'components/Petitions';
 import getPetitions from 'selectors/petitions';
 
-const PetitionsContainer = React.createClass({
+const PetitionsContainer = withRouter(React.createClass({
   componentWillMount () {
     this.props.fetchPetitions(this.props);
   },
 
+  componentWillReceiveProps (nextProps) {
+    if (this.props.location.pathname !== nextProps.location.pathname) {
+      this.props.fetchPetitions(nextProps);
+    }
+  },
+
+  getAutocompleteProps () {
+    const router = this.props.router;
+    const updateCityFilterValue = this.props.updateCityFilterValue;
+    const cityFilterValue = this.props.cityFilterValue;
+
+    return {
+      name: 'city-filter',
+      endpoint: 'cities',
+      suggestionFormatter: (suggestion) => {
+        if (suggestion.name && suggestion.zips) {
+          return suggestion.name + ' - ' + suggestion.zips[0];
+        }
+
+        return '';
+      },
+      getFormValue: (suggestion) => {
+        return suggestion;
+      },
+      suggestionsLimit: 4,
+      helper: {
+        value: {
+          data: cityFilterValue
+        },
+        onChange (newValue) {
+          updateCityFilterValue(newValue);
+
+          if (newValue.id) {
+            router.push(petitionsPath({city: newValue}));
+          } else {
+            router.push(petitionsPath());
+          }
+        },
+        onBlur () {}
+      },
+      html: {
+        placeholder: 'Filter by city'
+      }
+    };
```

Comment:
This is already what‘s being done.
The location changes, but this doesn‘t automatically trigger a new fetchPetitions, as `componentWillMount` is not called again. If I directly make the `fetchPetitions` call in the onChange, the `props.location` which is used in it is still the old value, which would display the same petitions again.
That‘s why I put it in `componentWillReceiveProps` and do the check to see if the location changed there. I don‘t think there was a more suited event.